### DISCUSSION TITLE
How to replace the deprecated PARTIAL syntax (DQL)?

### DISCUSSION THREAD

**Discussion by kevinpapst** [doctrine/doctrine2#11034] [Support Questions]
After recent composer updates (in a Symfony project) I am hit by:

> User Deprecated: PARTIAL syntax in DQL is deprecated. (Parser.php:1850 called by Parser.php:2281, https://github.com/doctrine/orm/issues/8471, package doctrine/orm)

Unfortunately the linked issue does not explain how to replace PARTIAL queries.

Why most of use PARTIAL was explained years ago by Marco:
https://ocramius.github.io/blog/doctrine-orm-optimization-hydration/

My project uses this technique for 80% of all object trees that needs to be loaded.  Here is a very simple example:
https://github.com/kimai/kimai/blob/main/src/Repository/Loader/InvoiceLoader.php#L43

In my case there are pages where PARTIAL fetches replace 10+ joins. Without PARTIAL I have literally no idea how to run the application.

My question is: is there a forward compatible replacement, that still uses the ORM? How can I replace these calls?

EDIT: Seems this is work in progress. There are many open & closed issues about this topic. 
For everyone else wondering where to get more infos, these look like a good follow: 
- https://github.com/doctrine/orm/issues/10314 
- https://github.com/doctrine/orm/pull/8391

**Comment by beberlei** [comment]
Please keep discussion about this topic on #8471

**Reply by kevinpapst** [reply]
In a closed issue, so nobody will see it? That is really weird

**Reply by beberlei** [reply]
The issue is linked from the deprecation in the code and is the first thing everyone will see and click.
